### DISCUSSION TITLE
Support Loading levels from https://maticzpl.xyz/lbpfind/ within game.

### DISCUSSION THREAD

**Discussion by Tydium** [LBPUnion/ProjectLighthouse#1017] [Ideas]
I recently had discovered that "LittleBigRefresh" had managed to incorporate this onto their custom server. It would AWESOME if we had something similar. We can have a command like they do in game and also have something on beacons website for it as well for logged in players. It would DEFINITELY EXPAND the amount of levels accessible to beacon through this feature (would also help players replay their old levels that may have been archived here such as my own). I don't expect this would be easy to implement, so I wouldn't expect it anytime soon ofc. However if its possible then maybe in the far future :)

This would save a huge hassle of needing users to even upload levels to Lighthouse in the first place too. A less "technically capable" user would be easily able to play ANY old level they wish to play. I know the argument of "just ask a player to upload your level request", but this would COMPLETELY eliminate even needing to ask anyone.

![image](https://github.com/LBPUnion/ProjectLighthouse/assets/67938521/256ac664-2d28-456e-8435-ed2840bf045d)

**Comment by Tydium** [comment]
Im unsure exactly, how LittleBigRefresh does it, but it seems they just make the level name the hash. Im pretty sure the website also provides the Level Name, Description, and Icon. So hopefully it would help  :)

**Comment by m88youngling** [comment]
I'm going to take a personal vote _against_ this suggestion (at least for Beacon). If this was to be implemented in Project Lighthouse or Project Lighthouse 2, there should be a configuration option for an administrator to disallow this feature. The archive has plenty of inappropriate content that is impossible to review in its entirety, and this provides a way for the _server_ to load the content from the archive for the user, as opposed to the user pulling the archive content first and then uploading it to the Lighthouse server. This means that the server administrators would then be responsible for downloading potentially inappropriate content (or even CSAM, there could be that in the archive and that would be disastrous for the server to hand over to users)

If it was to be implemented, it should come down to the administrator of the server to decide whether the feature should be turned on, and I'd argue against turning it on for Beacon. Having each user be responsible for levels they pull from the archive is one thing, but I have yet to be convinced that it would be a good idea for a server to fully provide content from the archive unmoderated. It could have serious legal and safety consequences, and it's important for server administrators to minimize their liability. The last thing I want, personally, is for a service like Beacon to fetch CSAM content out of the archive if it exists and then hand it over to a user. 

The user should use their own tools or another service to download the level from the archive and upload it to the server. If it's inappropriate for the server, then the server administrators moderate the content. In this situation, it's easier to prescribe responsibility onto the user rather than on the server itself, which is much easier to manage. Because of the risk involved, I don't recommend that this get implemented, and if it does, my personal opinion is that server administrator should only turn the feature on if they're running a private instance of Lighthouse with people they trust.

**Comment by Tydium** [comment]
Thank you for sharing your detailed thoughts on the proposal to support loading levels from the archive. Your concerns about the potential risks, especially regarding inappropriate content, are valid and important to address. However, I believe that with the right safeguards and considering some additional points, implementing this feature could greatly enhance user experience and accessibility. Here are some points to consider:

### Enhanced User Experience and Accessibility:
- Integrating the ability to load levels directly within the game simplifies the process for users. This convenience can significantly enhance the user experience by reducing the steps required to access new content, thereby increasing engagement and satisfaction.

### Temporary Data Loading:
- It’s important to note that the proposed implementation would load the data from the archive temporarily. The levels would not be integrated or permanently stored in Beacon’s system. This temporary nature reduces the risk of Beacon inadvertently hosting inappropriate content. Additionally, interactions such as commenting or hearting levels wouldn’t be saved, making this feature more of a transient viewing tool.

### Implementation of Robust Moderation Tools:
- To mitigate the risks associated with inappropriate content, we can develop and implement robust moderation tools. Automated content filtering systems and community reporting mechanisms can help ensure that any harmful or inappropriate content is identified and removed promptly. We can make some type of system to ban certain hashes from ever being loaded. If the user would want to load them, like you said they would need to host their own private server for it.

### Administrator Control and Configurability:
- I agree with your suggestion that there should be a configuration option for server administrators to enable or disable this feature. This flexibility ensures that each administrator can make decisions based on their specific community and risk tolerance. For instance, an administrator could choose to allow this feature only in private or well-moderated servers.

### Community and User Responsibility:
- Empowering the community to assist in moderation can be an effective strategy. By involving users in the reporting and moderation process, we can distribute the responsibility and ensure quicker identification of inappropriate content. This collaborative approach can significantly reduce the burden on server administrators. I feel that "us" as a community should help ban the hashes that aren't safe for playing at all. As a Beacon player myself I also wouldn't want to find NSFW content as I'm playing. We should encourage each other and the community to clean up as much as we can.

### Benefits to the Project and The LittleBigPlanet Community:
- Implementing this feature can serve as a competitive advantage, setting Project Lighthouse apart from other similar private servers by offering a seamless and integrated content loading experience. This innovation could attract a larger user base and foster a more active and engaged community. 

In conclusion, while the risks you mentioned are real, they can be managed through thoughtful implementation of moderation tools, configurability, and legal safeguards. Considering that the levels are not permanently uploaded to Beacon and the feature mainly acts as a backend temporary loader, we can address safety concerns effectively. By doing so, we can create a safer and more enjoyable experience for all users, ultimately benefiting the entire community. 

Also to serve as a rather "sad" reminder. The "Official" LittleBigPlanet servers are gone. Nothing else remains. This feature honors the legacy of the original LittleBigPlanet servers, providing a vital link to users' cherished content. That archive rightfully belongs to us as a community. Being able to access the content freely without struggle would be an absolute blessing and godsend to reviving what we once lost.

**Comment by jackcaver** [comment]
To even "temporarily" host the resources required for the level, beacon has to have this content locally or on a remote server where it can download it, which means that the server admins has to host it somewhere, so the responsibility of hosting potentially unsafe content doesn't go away, also reading a big SQLite database could slow down the server while it will be loading the level and it's resources for the specified level
